<?xml version="1.0"?>
<launch>
    <arg name="mono3d_flag" default="true"/>
    <arg name="mono3d_ckpt_file" default="/home/itachi/project/catkin_ws_test/src/ros2_vision_inference/models/mono3d_yolox_576_768.onnx"/>

    <arg name="seg_flag" default="false"/>
    <arg name="seg_ckpt_file" default="/home/itachi/project/catkin_ws_test/src/ros2_vision_inference/models/bisenetv1.onnx"/>
    
    <arg name="monodepth_flag" default="false"/>
    <arg name="monodepth_ckpt_file" default="/home/itachi/project/catkin_ws_test/src/ros2_vision_inference/models/monodepth_res101_384_1280.onnx"/>
    
    <arg name="camera_topic" default="/kitti/camera_color_left/image_raw"/>
    <arg name="camera_param" default="/kitti/camera_color_left/camera_info"/>
    <arg name="bbox_topic" default="/bboxes"/>

    <node pkg="ros2_vision_inference" type="ros2_vision_inference.py" name="vision_inference_node" output="screen">
        <param name="MONO3D_FLAG" value="$(arg mono3d_flag)"/>
        <param name="MONO3D_CKPT_FILE" value="$(arg mono3d_ckpt_file)"/>

        <param name="SEG_FLAG" value="$(arg seg_flag)"/>
        <param name="SEG_CKPT_FILE" value="$(arg seg_ckpt_file)"/>

        <param name="MONODEPTH_FLAG" value="$(arg monodepth_flag)"/>
        <param name="MONODEPTH_CKPT_FILE" value="$(arg monodepth_ckpt_file)"/>


        <remap from="/image_raw" to="$(arg camera_topic)"/>
        <remap from="/camera_info" to="$(arg camera_param)"/>
        <remap from="/bboxes" to="$(arg bbox_topic)"/>
    </node>

</launch>